src/window/window_wrapper/keyboard_manager.rs

```diff
@@ -157,7 +158,9 @@ impl KeyboardManager {
                         if key_event.state == ElementState::Pressed {
                             // Determine if this key event represents a key which won't ever
                             // present text.
-                            if let Some(key_text) = is_control_key(key_event.logical_key) {
+                            if let Some(key_text) =
+                                is_control_key(self.shift, key_event.logical_key)
```

Comment:
I don't think this is a good idea, we're basically replacing shift with tab.

But, to make it right - we need some refactoring.

```rs
if key_event.state == ElementState::Pressed {
   ...
}
```

What all code in this block is doing, is - formatting keybinding, if it looks like keybinding.

Most of the code - if's, and calling `format_keybinding_string` with different args.

This needs to be refactored into something like this:

```rs
if let Some(keybinding) = self.maybe_format_keybinding(key_event) {
  ... send keybinding
}
```

`maybe_format_keybinding` should use guard clauses to return None for keybindigs which should be ignored, and Some keybinding for valid cases.

@smolck what do you think?

@Kethku do you like this idea?

Reply:
I'm thinking about this. I don't have a good model in my head for what needs to happen here, so I'm going to hold off a bit and come back to it. I think we need more information from upstream in order to really do this correctly.